### PR TITLE
Mongodb Context Skeleton + Mongo Context/Model Tests

### PR DESCRIPTION
**Pull request by townerr**
Moved the db context to a testing branch which contains skeleton code for the context and tests for its methods.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by SukafuDesu — APPROVED** [review]

**Review by escalonn — CHANGES_REQUESTED** [review]
the branch is not up to date, fix that

File: FakebookNotifications/FakebookNotifications.Testing/FakebookNotifications.Testing.csproj [hunk 1]
```diff
@@ -23,6 +22,7 @@
 
 	<ItemGroup>
 		<ProjectReference Include="../FakebookNotifications.WebApi/FakebookNotifications.WebApi.csproj" />
+    <ProjectReference Include="..\FakebookNotifications.DataAccess\FakebookNotifications.DataAccess.csproj" />
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.Testing/FakebookNotifications.Testing.csproj]: formatting

File: FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+            //Arrange
+            //Setup Context Settings
+            _mockSettings.Setup(s => s.Value).Returns(settings);
+
+            //Setup Mongo Client
+            _mockClient.Setup(c => c.GetDatabase(_mockSettings.Object.Value.DatabaseName, null))
+                .Returns(_mockDB.Object);
+
+            //Act
+            var context = new NotificationsContext(_mockSettings.Object); //create context and test constructor
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs:38]: what does the mongoclient have to do with this test?

File: FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+            //Assert
+            Assert.NotNull(context);
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs]: ctors never return null, so just have an empty 'assert' section with a comment indicating success if no exception

File: FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        //Constructor to initialize mocked components before each test
+        public MongoDbTests()
+        {
+            _mockSettings = new Mock<IOptions<NotificationsDatabaseSettings>>();
+            _mockDB = new Mock<IMongoDatabase>();
+            _mockClient = new Mock<IMongoClient>();
+
+            settings = new NotificationsDatabaseSettings()
+            {
+                ConnectionString = "mongodb://localhost:27017",
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs]: why are we providing a value here that seems to depend on a db running for our unit tests? or if this value isnt being used, then why set it?

File: FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+        [Fact]
+        public void UserModel_IdTest()
+        {
+            //Arrange
+            var user = new Mock<User>();
+            user.Object.Id = "123123213213";
+
+            //Assert
+            Assert.Equal("123123213213", user.Object.Id);
+        }
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.Testing/MongoDbTests.cs]: these tests have several problems.
- any new test in a tdd-fail PR is supposed to fail
- mocking the class you're testing
- arrange/act steps misidentified
- but we don't need to unit test POCO classes anyway.

File: FakebookNotifications/FakebookNotifications.WebApi/Startup.cs [hunk 6]
```diff
@@ -27,6 +31,12 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            //DB settings configuration to get from appsettings
+            services.Configure<NotificationsDatabaseSettings>(
+                Configuration.GetSection(nameof(NotificationsDatabaseSettings)));
+
+            // Create context and repos services
+            services.AddScoped<NotificationsContext>();
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.WebApi/Startup.cs]: this code is not relevant to the new tests & doesn't belong in this PR

File: FakebookNotifications/FakebookNotifications.WebApi/appsettings.json [hunk 7]
```diff
@@ -1,4 +1,10 @@
 {
+  "NotificationsDatabaseSettings": {
+    "UserCollection": "User",
+    "NotificationsCollection": "Notifications",
+    "ConnectionString": "mongodb://localhost:27017",
+    "DatabaseName": "NotificationsDb"
+  },
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.WebApi/appsettings.json]: also not relevant to this PR

File: FakebookNotifications/FakebookNotifications.DataAccess/Models/Notification.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
+﻿using MongoDB.Bson;
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.DataAccess/Models/Notification.cs:1]: if this code was going to be unit tested, it'd need to have `get => throw new NotImplementedException()` for all these so that the tests fail. but classes like this with only auto-properties don't need to be unit tested.
(hence these classes should be added in an unlabelled PR, not this PR, but i'll overlook that)

**Review by escalonn — CHANGES_REQUESTED** [review]

**Review by escalonn — APPROVED** [review]
